File: archive/msconvert.pl
#!/usr/bin/env perl

use strict;
use warnings;
use Net::FTP;
use File::Path qw(make_path remove_tree);

my $ftp = Net::FTP->new("massive.ucsd.edu", Debug => 0)
  or die "Cannot connect to host: $@";

#	Apparently NEED to login when scripting
$ftp->login("anonymous",'-anonymous@')
	or die "Cannot login ", $ftp->message;

my $local_root  = "/Users/jakewendt/";
my $remote_root = "/";
my $base = "MSV000079053";
my $local_base  = $local_root.$base;
my $remote_base = $remote_root.$base;

$ftp->cwd($remote_base."/raw")
	or die "Cannot change working directory ", $ftp->message;
my @bacteria = $ftp->ls;

if( ! -d $local_base."/raw" ) {
	make_path( $local_base."/raw" ) or die "Failed make_path $!";
}

if( ! -d $local_base."/sequence" ){
	make_path( $local_base."/sequence" ) or die "Failed make_path $!";
}

if( ! -d $local_base."/out" ){
	make_path( $local_base."/out" ) or die "Failed make_path $!";
}


#	http://fields.scripps.edu/rawconv/download/MSFileReader%202.2.62.zip

my $msconvert = 'C:\Program Files\ProteoWizard\ProteoWizard 3.0.18187.b51377ef8\msconvert.exe';

foreach (@bacteria) {
	my $bacterium = $_;
	print "$bacterium\n";

	$ftp->cwd($remote_base."/raw/".$bacterium)
		or die "Cannot change working directory ", $ftp->message;
	my @raws = $ftp->ls;

	foreach(@raws) {
		my $raw = $_;
		print "$raw\n";
		chdir $local_base."/raw/";
		mkdir $bacterium if( ! -d $bacterium );
		chdir $bacterium;

		$ftp->cwd($remote_base."/raw/".$bacterium)
			or die "Cannot change working directory ", $ftp->message;
		if( ! -e $raw ){
			$ftp->get($raw)
				or die "get raw file failed ", $ftp->message;
		}

		#	lots of quotes are needed
		print `"$msconvert" $raw --mgf --filter "msLevel 2" --filter "zeroSample removeExtra"`;


		$ftp->cwd($remote_base."/sequence/".$bacterium)
			or die "Cannot change working directory ", $ftp->message;

		my @sequences = $ftp->ls;
#		die "Not just 1 sequence found." if $#sequences != 0;
#		Some have more than 1 sequence.
#		die "No sequence found." if $#sequences < 0;

		foreach(@sequences) {
			my $sequence = $_;
			print "$sequence\n";

			chdir $local_base."/sequence/";
			mkdir $bacterium if( ! -d $bacterium );
			chdir $bacterium;

			$ftp->cwd($remote_base."/sequence/".$bacterium)
				or die "Cannot change working directory ", $ftp->message;
			if( ! -e $sequence ){
				$ftp->get($sequence)
					or die "get sequence failed ", $ftp->message;
			}


			#	MODa

#print `ls -ali`;



		}	#	foreach(@sequences)

	}	#	foreach(@raws) {
}

$ftp->quit;

File: archive/msconvert.rb
#!/usr/bin/env ruby

require 'parallel'

require 'net/ftp'
#https://ruby-doc.org/stdlib-2.4.0/libdoc/net/ftp/rdoc/Net/FTP.html

require 'fileutils'

ftp = Net::FTP.new('massive.ucsd.edu')
ftp.login
puts "Logged in"

local_root  = "/Users/jakewendt/"
remote_root = "/"
base = "MSV000079053"
local_base  = "#{local_root}#{base}"
remote_base = "#{remote_root}#{base}"

FileUtils.mkdir_p "#{local_base}/raw" unless File.directory? "#{local_base}/raw"
FileUtils.mkdir_p "#{local_base}/sequence" unless File.directory? "#{local_base}/sequence"
FileUtils.mkdir_p "#{local_base}/out" unless File.directory? "#{local_base}/out"


puts "chdir #{remote_base}/raw"
ftp.chdir("#{remote_base}/raw")



#
##	http://fields.scripps.edu/rawconv/download/MSFileReader%202.2.62.zip
#
#my $msconvert = 'C:\Program Files\ProteoWizard\ProteoWizard 3.0.18187.b51377ef8\msconvert.exe';
#
##	parallelize like ... syryu/scripts/modification_locator.rb
#



ftp.nlst.each do |bacterium|
#Parallel.each(ftp.nlst) do |bacterium|
	puts bacterium

	ftp.chdir("#{remote_base}/raw/#{bacterium}")

#	Parallel.each(ftp.nlst) do |raw|
	ftp.nlst.each do |raw|
		puts "-#{raw}"

		FileUtils.mkdir_p "#{local_base}/raw/#{bacterium}" unless File.directory? "#{local_base}/raw/#{bacterium}"
		Dir::chdir "#{local_base}/raw/#{bacterium}"

		ftp.chdir("#{remote_base}/raw/#{bacterium}")

		if File.exists? raw
			puts "-#{raw} already exists"
		else
			puts "-Getting #{raw}"
			ftp.get(raw)
		end



		puts "-Running msconvert on #{raw}"

#		#	lots of quotes are needed
#		print `"$msconvert" $raw --mgf --filter "msLevel 2" --filter "zeroSample removeExtra"`;






		ftp.chdir("#{remote_base}/sequence/#{bacterium}")
		ftp.nlst.each do |sequence|

			FileUtils.mkdir_p "#{local_base}/sequence/#{bacterium}" unless File.directory? "#{local_base}/sequence/#{bacterium}"
			Dir::chdir "#{local_base}/sequence/#{bacterium}"

			ftp.chdir("#{remote_base}/sequence/#{bacterium}")

			if File.exists? sequence
				puts "--#{sequence} already exists"
			else
				puts "--Getting #{sequence}"
				ftp.get(sequence)
			end





			puts "--Creating config"


			puts "--Running MODa on #{raw} and #{sequence}"




		end	#	ftp.nlst.each do |sequence|

	end	#	ftp.nlst('*').each |raw|

end	#	ftp.nlst('*').each do |bacterium|


ftp.close
puts "Closed"
